lib/src/main/scala/spinal/lib/eda/symbiflow/yosys/yosys.scala

```diff
@@ -0,0 +1,297 @@
+  */
+case class Input(file: Path, frontend: String, opt: String*)
+    extends YosysScript {
+  override def toString(): String =
+    s"""read_${frontend} ${opt.mkString(" ")} ${file.normalize.toString}"""
+}
+
+/** Class that rappresent a yosys backend command
+  *
+  * @param file the file name or path
+  * @param backend the backend to use when writing the output file
+  * @param opt the parameters for the yosys frontend
+  */
+case class Output(file: Path, backend: String, opt: String*)
+    extends YosysScript {
+  override def toString(): String =
+    s"""write_${backend} ${opt.mkString(" ")} ${file.normalize.toString}"""
+
+    def outputFolder(path: Path): Output = Output(file=path.resolve(file),backend,opt:_*)
+}
+
+object Yosys {
+  def loadSystemVerilog[T <: Component](report: SpinalReport[T]): Yosys = {
+    val command = Yosys()
+    report.rtlSourcesPaths.foreach(x => command.addInputFile(Paths.get(x), "verilog", "-sv"))
+    command + setTop(report.toplevelName)
+  }
+
+  def loadVerilog[T <: Component](report: SpinalReport[T]): Yosys = {
+    val command = Yosys()
+    report.rtlSourcesPaths.foreach(x => command.addInputFile(Paths.get(x), "verilog"))
+    command + setTop(report.toplevelName)
+  }
+
+  def setTop(top: String): Yosys = {
+    val command = Yosys()
+    command.addCommand("prep", "-top", top)
+    command
+  }
+
+  def load_verilog(): Yosys = {
+    val command = Yosys()
+    val verilogFiles = command.getAllPrerequisiteFromExtension(".*.v")
+    verilogFiles.foreach(command.addInputFile(_, "verilog"))
+    command
+  }
+
+  def export_smt2(file: String = "model.smt2"): Yosys = {
+    val ret = Yosys()
+    ret.addOutputFile(Paths.get(file), "smt2")
+    ret
+  }
+
+  def model(mode: String,
+            multiclock: Boolean = false,
+            memoryMap: Boolean = false): Yosys = {
+    val modelGen = Yosys()
+    if (memoryMap) modelGen.addCommand("memory_map")
+    else modelGen.addCommand("memory_nordff")
+    modelGen.addCommand("memory_nordff")
+    if (multiclock) {
+      modelGen.addCommand("clk2fflogic")
+    } else {
+      modelGen.addCommand("async2sync")
+      modelGen.addCommand("chformal", "-assume", "-early")
+    }
+    if (mode == Mode.bmc || mode == Mode.prove)
+      modelGen.addCommand("chformal", "-live", "-fair", "-cover", "-remove")
+    if (mode == Mode.cover)
+      modelGen.addCommand("chformal", "-live", "-fair", "-remove")
+    if (mode == Mode.live) {
+      modelGen.addCommand("chformal", "-assert2assume")
+      modelGen.addCommand("chformal", "-cover", "-remove")
+    }
+    modelGen.addCommand("opt_clean")
+    modelGen.addCommand("setundef", "-anyseq")
+    modelGen.addCommand("opt", "-keepdc", "-fast")
+    modelGen.addCommand("check")
+    modelGen.addCommand("hierarchy", "-simcheck")
+    modelGen
+  }
+
+  def synthesize(target: String): Yosys = {
+    val command = Yosys()
+    command.addCommand("synth_" + target)
+    command.addOutputFile("syntetized.json", "json")
```

Comment:
Please don't hardcode the output file type to JSON. In my current setup, I run Yosys with `synth_xilinx -top MyTop -edif output.edif` so I can pipe it directly into Vivado.

Reply:
I agree, I should not force an output in this command.